src/auth/PrivateRoute.tsx

```diff
@@ -21,18 +22,24 @@ const PrivateRoute: React.FC<PrivateRouteProps> = ({
   exact,
   sensitive,
   strict,
-}: PrivateRouteProps) => (
-  <Route
-    component={withAuthenticationRequired(component, {
-      returnTo: path,
-      onRedirecting: () => <div>Redirecting...</div>,
-    })}
-    path={path}
-    exact={exact}
-    sensitive={sensitive}
-    strict={strict}
-  />
-)
+}: PrivateRouteProps) => {
+  const { user } = useAuth0()
+  const route: React.ComponentType<any> = user?.email_verified
```

Comment:
I had to logout then login after I verified my email to get through this. I think it's because I made auth0 to [store cache to localstorage](https://github.com/CruzHacks/cruzhacks-2022-website/blob/ed892ec7518baa77282a121b827a29359d8fdd64/src/auth/Auth0ProviderWithHistory.tsx#L34). Perhaps we don't need it to store cache, or can we refresh them somehow?

Reply:
With auth0 it's caching in localstorage since that's how we have it configured. We can do either, make an endpoint that get's the user verified status when localstorage is false to double verify. Another option is to just ask people to login and logout after their first login (once verified).  Another option is just to use session storage or store it in state variables and change the auth context. The downside of this approach is after page is on refresh this won't save the current page state and the user would have to relogin

Reply:
Looks like you went with the last approach?